### ISSUE TITLE
Scalene: Program did not run for long enough to profile.

### ISSUE THREAD

**Issue by jey07** [plasma-umass/scalene#613]
I tried to execute below function from google collab but it says that :

"Scalene: Program did not run for long enough to profile"

`def function():
  dict_main = {}
  for idx in range(df_data.shape[0]):
      row = {}
      for column in list(df_data.columns):
          if pd.isna(df_data[column].iloc[idx]):
              row[str(column)] = None
          else:
              row[str(column)] = df_data[column].iloc[idx]
      dict_main[str(idx)] = row`

**Comment by KyleStiers** [comment]
I'm also getting this, despite the python running for several minutes. Is there a way to ask for scalene to run regardless with less accuracy?

I'm running py3.10

**Comment by JulienPalard** [comment]
I can reproduce it with:

```python
from decimal import Decimal, localcontext
from itertools import count
from functools import cache

@cache
def fib(n):
    if n in (0, 1):
        return 1
    x = n // 2
    return fib(x - 1) * fib(n - x - 1) + fib(x) * fib(n - x)

@cache
def approx_phi(n):
    return Decimal(fib(n + 1)) / Decimal(fib(n))

def approx_phi_up_to(n_digits):
    with localcontext() as ctx:
        ctx.prec = n_digits + 1
        for n in count():
            step1 = approx_phi(2 ** n)
            step2 = approx_phi(2 ** (n+1))
            if step1 == step2:
                return step1

import sys

if __name__ == "__main__":
    print(approx_phi_up_to(int(sys.argv[1])))
```

Running : `time scalene phi.py 300000` gives:

```
Scalene: Program did not run for long enough to profile.

real	0m35.697s
user	0m35.514s
sys	0m0.115s
```

**Comment by emeryberger** [comment]
Does not reproduce on Python 3.10.11, Mac, Scalene version 1.5.21.4. Please provide more system info - how did you install it? `pip` or `conda`? Etc. Thanks in advance!

**Comment by JulienPalard** [comment]
Thanks a lot for taking the time trying to reproduce my issue, appreciated, I'm sorry I failed to provide a proper reproducer in the first place :(

looks like I have a different behavior if the tested file is in a directory called `include`:
```shell
$ cd /tmp
$ mkdir test
$ cd test
$ python -m venv .venv
$ source .venv/bin/activate
$ pip install scalene
[...]
Successfully installed Jinja2-3.1.2 MarkupSafe-2.1.3 cloudpickle-2.2.1 markdown-it-py-3.0.0 mdurl-0.1.2 psutil-5.9.5 pygments-2.15.1 pynvml-11.4.1 rich-13.4.2 scalene-1.5.21.4 wheel-0.40.0
$ cat > phi.py
[...]
$ scalene phi.py 300000 >/dev/null
# It worked.
$ mkdir sub
$ mv phi.py sub/
$ scalene sub/phi.py 300000 >/dev/null
# It works too
$ mv sub include
$ scalene include/phi.py 300000 >/dev/null
Scalene: Program did not run for long enough to profile.
```

The presence of an `include/` directory is not enough, the tested file has to be in it.

To be sure I'm trying again:

```
$ tree include/ sub/
include/
└── phi.py
sub/
└── phi.py

2 directories, 2 files
$ sha1sum include/phi.py sub/phi.py 
8226ce173bf383a322abce3765c76b5938cf2974  include/phi.py
8226ce173bf383a322abce3765c76b5938cf2974  sub/phi.py
# Beware this is the version without `print(`...`)`.
$ scalene include/phi.py 30000
Scalene: Program did not run for long enough to profile.
$ scalene sub/phi.py 30000
# it works
$ scalene include/phi.py 30000
Scalene: Program did not run for long enough to profile.
$ scalene sub/phi.py 30000
# it works
```

**Comment by emeryberger** [comment]
Thanks for the report - I was able to reproduce the issue and fix it (I hope). Please try with the repo version (`python3 -m pip install git+https://github.com/plasma-umass/scalene`) - thanks!

**Comment by JulienPalard** [comment]
Yes it fixed my issue!

I don't know how well it was finally related to #613 though...

Thanks a lot @emeryberger!

PS: I had to `--force-reinstall`, you could add it the next time you provide this command in an issue.

PPS: Did you know you can make people test branches with `python3 -m pip install --force-reinstall git+https://github.com/plasma-umass/scalene@THE_BRANCH`? so you can make people test something before merging in the main branch. I like this feature a lot :))

**Comment by BrettskiPy** [comment]
I took a similar approach to resolve this issue.  I uninstalled `1.5.23`.
```
pip uninstall scalene
```
and installed the previous release
```
pip install scalene==1.5.22.2
```

**Comment by emeryberger** [comment]
Hm, that's unfortunate (since the bug fix was already incorporated into 1.5.23). Can you share any more details?

**Comment by BrettskiPy** [comment]
> Hm, that's unfortunate (since the bug fix was already incorporated into 1.5.23). Can you share any more details?

I'm on a Mac running python `3.10.4`. 

I ran the command `scalene -m pytest` and I was using the`@profile` decorator on a test function.

I just uninstalled `1.5.22.2` and ran `pip install scalene` and it is now working just fine with `1.5.23`  Perhaps I just had an older version in my venv 🤷‍♂️ Thank you for the quick response.

**Comment by emeryberger** [comment]
@aj-in we had an issue with building the various wheels, but this is now fixed. You should be able to load the latest Scalene just by `python3 -m pip install -U scalene`.
